3266 docs rfc rewrite migrate low downtime live migration as migrate with near zero downtime (#3327)
timescale/docs: migrate/live-migration.md

tags: [recovery, logical backup, replication]
---
import MigrationPrerequisites from "versionContent/_partials/_migrate_prerequisites.mdx";
import SourceTargetNote from "versionContent/_partials/_migrate_source_target_note.mdx";
import OpenSupportRequest from "versionContent/_partials/_migrate_open_support_request.mdx"
import DoNotRecommendForLargeMigration from "versionContent/_partials/_migrate_pg_dump_do_not_recommend_for_large_migration.mdx";
import MigrateSetupEnvironment from "versionContent/_partials/_migrate_live_setup_environment.mdx";
import MigrateSetupEnvironmentMST from "versionContent/_partials/_migrate_live_setup_environment_mst.mdx";
import MigrateSetupEnvironmentAWSRDS from "versionContent/_partials/_migrate_live_setup_environment_awsrds.mdx";
import MigrateDataToCloud from "versionContent/_partials/_migrate_live_migrate_data.mdx";
import MigrateAWSRDSConnectIntermediary from "versionContent/_partials/_migrate_awsrds_connect_intermediary.mdx";

# Live migration

You use the [live-migration][live-migration-docker-image] Docker image to move 100GB-10TB+ of data to a Timescale Cloud service 
seamlessly with only a few minutes downtime. 

> [live-migration][live-migration-docker-image] is an end-to-end solution that copies the database schema and data to 

Reviewer: ```suggestion
[Live-migration][live-migration-docker-image] is an end-to-end solution that copies the database schema and data to 
```

your target Timescale Cloud service, then replicates updates in your source database 

> to the target service in real-time. live-migration uses the Postgres logical replication 

Reviewer: ```suggestion
your target Timescale Cloud service, then replicates the database activity in your source database 
to the target service in real-time. Live-migration uses the Postgres logical decoding 
```

functionality and leverages [pgcopydb]. 

Best practice is to use live-migration when:
- Modifying your application logic to perform dual writes is a significant effort. 
- The insert workload does not exceed 20,000 rows per second, and inserts are batched.

  Use [Dual write and backfill][dual-write-and-backfill] for greater workloads.
- Your source database:
  - Uses `UPDATE` and `DELETE` statements on uncompressed time-series data.

>     live-migration does not support replicating `INSERT`/`UPDATE`/`DELETE` statements on compressed data.

Reviewer: ```suggestion
    Live-migration does not support replicating `INSERT`/`UPDATE`/`DELETE` statements on compressed data.
```
